### PR TITLE
OCPRHV-482 autopinning, hugepages in machineset YAML and install config params

### PR DESCRIPTION
**Pull request by emarcusRH**
enterprise-4.9
https://issues.redhat.com/browse/OCPRHV-482
Previews: 
https://deploy-preview-36333--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/installing-rhv-customizations?utm_source=github&utm_campaign=bot_dp#installation-configuration-parameters-additional-machine_installing-rhv-customizations

https://deploy-preview-36333--osdocs.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-rhv.html#machineset-yaml-rhv_creating-machineset-rhv

* DEV review done by Gal Z in PR https://github.com/openshift/openshift-docs/pull/35574
* QE reviewed and ack by Micahel Burman

### CODE REVIEW

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -1114,6 +1114,39 @@ This example defines no affinity groups:
...
+<machine-pool>:
+autoPinningPolicy:
+  description:
+    AutoPinningPolicy defines the policy to
+    automatically set CPU and NUMA.
+  enum:
+    - none
+    - resize_and_pin
+  type: string
+----
```
Comment (Gal-Zaidman) [inline modules/installation-configuration-parameters.adoc:1131]: Delete this yaml

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -1114,6 +1114,39 @@ This example defines no affinity groups:
...
+<machine-pool>:
+hugepages:
+  description:
+    Hugepages - the size (KiB) of a VM's hugepages.
+  enum:
+    - 2048
+    - 1048576
+  format: int32
+  type: integer
+----
```
Comment (Gal-Zaidman) [inline modules/installation-configuration-parameters.adoc:1148]: Delete this yaml

**Comment by Gal-Zaidman** [comment]
DEV didn't give any ack, on the prev PR I made comments on corrections that weren't implemented

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -1114,6 +1114,39 @@ This example defines no affinity groups:
...
+|String
```
Comment (Gal-Zaidman) [inline modules/installation-configuration-parameters.adoc:1149]: This is an integer

File: modules/installation-configuration-parameters.adoc [hunk 4]
```diff
@@ -1114,6 +1114,39 @@ This example defines no affinity groups:
...
+    automatically set CPU and NUMA.
+  enum:
+    - none
+    - resize_and_pin
+  type: string
+----
+
+|String
+|`<machine-pool>.platform.ovirt.hugepages`
+|Optional. Hugepages is the size in KiB for defining hugepages in a VM. Supported values: 2048 or 1048576.
```
Comment (Gal-Zaidman) [inline modules/installation-configuration-parameters.adoc:1135]: give a link to the RHV docs about hugepages

File: modules/installation-configuration-parameters.adoc [hunk 5]
```diff
@@ -1114,6 +1114,39 @@ This example defines no affinity groups:
       affinityGroupNames: []
 ----
 |String
+|`<machine-pool>.platform.ovirt.AutoPinningPolicy`
+| Optional. AutoPinningPolicy defines the policy to automatically set the CPU and NUMA settings including pinning to the host for the instance. When the field is omitted, the default is `none`. Supported values: `none`, `resize_and_pin`.
```
Comment (Gal-Zaidman) [inline modules/installation-configuration-parameters.adoc:1118]: give a link to the RHV docs about AutoPinningPolicy

File: modules/installation-configuration-parameters.adoc [hunk 6]
```diff
@@ -1114,6 +1114,39 @@ This example defines no affinity groups:
...
+
+[source,yaml]
+----
+<machine-pool>:
```
Comment (maxwelldb) [inline modules/installation-configuration-parameters.adoc:1122]: Is this a section of a YAML file, or the entire file? Use commented ellipses and verify indentation if former.  

Leading the example with `<machine-pool>` gives me pause. Could you avoid using a fill-in by contextualizing the example, e.g., "For example, for a compute machine pool..."?

File: modules/machineset-yaml-rhv.adoc [hunk 7]
```diff
@@ -104,7 +106,9 @@ $ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
 <18> Specify the name of the secret that holds the {rh-virtualization} credentials.
 
 <19> Optional: Specify the workload type for which the instance is optimized. This value affects the `{rh-virtualization} VM` parameter. Supported values: `desktop`, `server` (default), `high_performance`. `high_performance` improves performance on the VM, but there are limitations. For example, you cannot access the VM with a graphical console. For more information see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Configuring_High_Performance_Virtual_Machines_Templates_and_Pools[Configuring High Performance Virtual Machines, Templates, and Pools] in the _Virtual Machine Management Guide_.
-<20> A list of affinity group names that should be applied to the VMs. The affinity groups must exist in oVirt.
+<20> Optional: AutoPinningPolicy defines the policy that automatically sets CPU and NUMA settings, including pinning to the host for this instance. Supported values: `none`, `resize_and_pin`.
+<21> Optional: Hugepages is the size in KiB for defining hugepages in a VM. Supported values: 2048 or 1048576.
```
Comment (maxwelldb) [inline modules/machineset-yaml-rhv.adoc:110]: I believe that "2048" and "1048576" should be monospace, as they're values of config parameters here.

**Review by maxwelldb — COMMENTED** [review]
/lgtm

Left a couple of suggestions.

**Comment by Gal-Zaidman** [comment]
holding this PR till DEV and QE will approve it

**Comment by emarcusRH** [comment]
Not sure how this was suddenly closed?

**Comment by emarcusRH** [comment]
> /lgtm
> 
> Left a couple of suggestions.

Thank you

File: modules/installation-configuration-parameters.adoc [hunk 8]
```diff
@@ -1129,6 +1114,14 @@ This example defines no affinity groups:
       affinityGroupNames: []
 ----
 |String
+|`<machine-pool>.platform.ovirt.AutoPinningPolicy`
+| Optional. AutoPinningPolicy defines the policy to automatically set the CPU and NUMA settings including pinning to the host for the instance. When the field is omitted, the default is `none`. Supported values: `none`, `resize_and_pin`. For more information see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Setting_NUMA_Nodes[Setting NUMA Nodes] in the _Virtual Machine Management Guide_.
```
Comment (maxwelldb) [inline modules/installation-configuration-parameters.adoc:1118]:
```suggestion
| Optional. AutoPinningPolicy defines the policy to automatically set the CPU and NUMA settings, including pinning to the host for the instance. When the field is omitted, the default is `none`. Supported values: `none`, `resize_and_pin`. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Setting_NUMA_Nodes[Setting NUMA Nodes] in the _Virtual Machine Management Guide_.
```

File: modules/installation-configuration-parameters.adoc [hunk 9]
```diff
@@ -1129,6 +1114,14 @@ This example defines no affinity groups:
...
+
+|String
+|`<machine-pool>.platform.ovirt.hugepages`
+|Optional. Hugepages is the size in KiB for defining hugepages in a VM. Supported values: `2048` or `1048576`. For more information see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Configuring_Huge_Pages[Configuring Huge Pages] in the _Virtual Machine Management Guide_.
```
Comment (maxwelldb) [inline modules/installation-configuration-parameters.adoc:1122]:
```suggestion
|Optional. Hugepages is the size in KiB for defining hugepages in a VM. Supported values: `2048` or `1048576`. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Configuring_Huge_Pages[Configuring Huge Pages] in the _Virtual Machine Management Guide_.
```

File: modules/machineset-yaml-rhv.adoc [hunk 10]
```diff
@@ -104,7 +106,9 @@ $ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
...
-<20> A list of affinity group names that should be applied to the VMs. The affinity groups must exist in oVirt.
+<20> Optional: AutoPinningPolicy defines the policy that automatically sets CPU and NUMA settings, including pinning to the host for this instance. Supported values: `none`, `resize_and_pin`. For more information see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Setting_NUMA_Nodes[Setting NUMA Nodes] in the _Virtual Machine Management Guide_.
```
Comment (maxwelldb) [inline modules/machineset-yaml-rhv.adoc:109]:
```suggestion
<20> Optional: AutoPinningPolicy defines the policy that automatically sets CPU and NUMA settings, including pinning to the host for this instance. Supported values: `none`, `resize_and_pin`. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Setting_NUMA_Nodes[Setting NUMA Nodes] in the _Virtual Machine Management Guide_.
```

File: modules/machineset-yaml-rhv.adoc [hunk 11]
```diff
@@ -104,7 +106,9 @@ $ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
...
+<21> Optional: Hugepages is the size in KiB for defining hugepages in a VM. Supported values: `2048` or `1048576`. For more information see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Configuring_Huge_Pages[Configuring Huge Pages] in the _Virtual Machine Management Guide_.
```
Comment (maxwelldb) [inline modules/machineset-yaml-rhv.adoc:110]:
```suggestion
<21> Optional: Hugepages is the size in KiB for defining hugepages in a VM. Supported values: `2048` or `1048576`. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html-single/virtual_machine_management_guide/index#Configuring_Huge_Pages[Configuring Huge Pages] in the _Virtual Machine Management Guide_.
```

**Review by maxwelldb — APPROVED** [review]
@emarcusRH Still lgtm, although I can't help but doubt myself when I see anything that changes the stacks of conditionals in the install-config.yaml parameters module. :fearful: 

Since I'm in here, suggested some commas to take or leave.

**Pull request merged**